hsr: add support to HSR interface

ffmancera opened the pull request:
This PR includes two commits. One is fixing the IPv6 address flag API usage from netlink-packet-route and the other is adding support to HSR/PRP interface.

ffmancera commented:
Please, do not merge this. It requires https://github.com/rust-netlink/netlink-packet-route/pull/58 to be merged and a release on netlink-packet-route.

cathay4t commented:
Sorry, I have to put this PR on hold till `netlink-packet-route` ships new release which will take a long time as I am refactoring its code there.

ffmancera commented:
> Sorry, I have to put this PR on hold till `netlink-packet-route` ships new release which will take a long time as I am refactoring its code there.

Can we create a release there and then another after the refactoring? This is blocking the Nmstate feature too.. Is there a problem is we release `netlink-packet-route` now and then later we move to 1.0?

ffmancera committed:
hsr: add support HSR/PRP interface

Add support to HSR/PRP interface.

Integration test added.

Example:

```
- name: prp0
  iface_type: hsr
  state: up
  hsr:
    port1: eth1
    port2: eth2
    supervision_addr: 01:15:4e:00:01:2d
    seq_nr: 48104
    multicast_spec: 0
    version: 0
    protocol: prp
```

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>
Signed-off-by: Gris Ge <fge@redhat.com>
